### PR TITLE
feat: creating task in task manager

### PR DESCRIPTION
**Pull request by axel7083**
Fixes https://github.com/projectatomic/studio-extension/issues/151

### CODE REVIEW

**Comment by feloy** [comment]
Where are the progress/error expected to be visible? I cannot see anything on the UI or tray on mac

**Comment by axel7083** [comment]
I had an error where the promise was not properly waited, I fixed it @feloy thanks for testing

Here is a demo

https://github.com/projectatomic/studio-extension/assets/42176370/081e4126-5fa5-4b92-b182-7b8c781cb729

**Comment by feloy** [comment]
Can you see the error displayed, in case of error?

I didn't tested on this PR but on my PR about deleting models, using similar code, and I was not able to see the error message appearing anywhere after calling `progress.report({ message: `Error: ${String(error)}` });`

**Comment by axel7083** [comment]
> Can you see the error displayed, in case of error?
> 
> I didn't tested on this PR but on my PR about deleting models, using similar code, and I was not able to see the error message appearing anywhere after calling `progress.report({ message: `Error: ${String(error)}` });`

Me neither, that's weird

**Comment by axel7083** [comment]
Okey @feloy I investigate, the bug is coming from the podman-core. I will open an issue, I already have a fix

File: packages/backend/src/studio-api-impl.ts [hunk 1]
```diff
@@ -73,9 +72,14 @@ export class StudioApiImpl implements StudioAPI {
     const model = await this.getModelById(modelId);
 
     // Do not wait for the pull application, run it separately
-    this.applicationManager.pullApplication(recipe, model).catch((error: unknown) => console.warn(error));
-
-    return Promise.resolve(undefined);
+    void podmanDesktopApi.window
```
Comment (lstocchi) [inline packages/backend/src/studio-api-impl.ts:75]: can we remove the `void` here?

**Review by lstocchi — APPROVED** [review]
Tested and works fine. 🚀  I'll just remove that void before the function call.

File: packages/backend/src/studio-api-impl.ts (hunk 1 above)
Comment (axel7083) [reply packages/backend/src/studio-api-impl.ts:75]: No the reason is avoid linter to complain that we have "dangling promises"

Comment (lstocchi) [reply packages/backend/src/studio-api-impl.ts:75]: for me it runs fine without it 🤔 
Do you mean the `yarn lint:check`?

Comment (benoitf) [reply packages/backend/src/studio-api-impl.ts:75]: I agree here with @lstocchi we shouldn't use that void to fake the compiler

also you're doing a catch so linter should not report any problem

you can also remove the `<void>` after `.withProgress<void>` as default is void

Comment (benoitf) [reply packages/backend/src/studio-api-impl.ts:75]: the rule is that you should use:

```ts
await promise // valid
promise.....catch(error...) //valid
```

and not
```ts
promise // invalid (and do not add void before to disable the linter as then at runtime you may have unhandled exception
void promise // no
```

**Pull request merged**